crates/uv/src/commands/toolchain/uninstall.rs

```diff
@@ -0,0 +1,120 @@
+    printer: Printer,
+) -> Result<ExitStatus> {
+    if preview.is_disabled() {
+        warn_user_once!("`uv toolchain uninstall` is experimental and may change without warning.");
+    }
+
+    let toolchains = InstalledToolchains::from_settings()?.init()?;
+
+    let requests = targets
+        .iter()
+        .map(|target| ToolchainRequest::parse(target.as_str()))
+        .collect::<Vec<_>>();
+
+    let download_requests = requests
+        .iter()
+        .map(|request| PythonDownloadRequest::from_request(request.clone()))
+        .collect::<Result<Vec<_>, downloads::Error>>()?;
+
+    let installed_toolchains: Vec<_> = toolchains.find_all()?.collect();
+    let mut matching_toolchains = BTreeSet::default();
+    for (request, download_request) in requests.iter().zip(download_requests) {
+        writeln!(
+            printer.stderr(),
+            "Looking for installed toolchains matching {request} ({download_request})"
+        )?;
+        let mut found = false;
+        for toolchain in installed_toolchains
+            .iter()
+            .filter(|toolchain| download_request.satisfied_by_key(toolchain.key()))
+        {
+            found = true;
+            if matching_toolchains.insert(toolchain) {
+                writeln!(
+                    printer.stderr(),
+                    "Found toolchain '{}' that matches {request}",
+                    toolchain.key()
+                )?;
+            }
+        }
+        if !found {
+            writeln!(printer.stderr(), "No toolchains found matching {request}")?;
+        }
+    }
+
+    if matching_toolchains.is_empty() {
+        if matches!(requests.as_slice(), [ToolchainRequest::Any]) {
+            writeln!(printer.stderr(), "No installed toolchains found",)?;
+        } else if requests.len() > 1 {
+            writeln!(
+                printer.stderr(),
+                "No toolchains found matching the requests"
+            )?;
+        } else {
+            writeln!(printer.stderr(), "No toolchains found matching the request")?;
+        }
+        return Ok(ExitStatus::Failure);
+    }
+
+    let tasks = futures::stream::iter(matching_toolchains.iter())
+        .map(|toolchain| async {
+            (
+                toolchain.key(),
+                fs_err::tokio::remove_dir_all(toolchain.path()).await,
+            )
+        })
+        .buffered(4);
+
+    let results = tasks.collect::<Vec<_>>().await;
+    let mut failed = false;
+    for (key, result) in results.iter().sorted_by_key(|(key, _)| key) {
+        if let Err(err) = result {
+            failed = true;
+            writeln!(
+                printer.stderr(),
+                "Failed to uninstall toolchain '{key}': {err}"
+            )?;
+        } else {
+            writeln!(printer.stderr(), "Uninstalled toolchain '{key}'")?;
+        }
+    }
+
+    if failed {
+        if matching_toolchains.len() > 1 {
+            writeln!(printer.stderr(), "Uninstall of some toolchains failed",)?;
```

Comment:
Extra `,` here sorry.

Reply:
I feel failed by my tooling.

Reply:
This one annoys me so much (that it's not automatic).

Reply:
Yeah. I'll need to watch for it I guess. Why would anyone want to keep that comma...